src/coreclr/vm/class.cpp

```diff
@@ -2210,7 +2210,7 @@ void MethodTable::DebugRecursivelyDumpInstanceFields(LPCUTF8 pszClassName, BOOL
 
     EX_TRY
     {
-        StackSString ssBuff;
+        StackSString ssBuff(SString::Utf8);
```

Comment:
What is the SString constructor that this ends up calling? What is the effect of the `SString::Utf8` argument?

Reply:
Boo. This is a mistake. I likely was checking the API and missed it didn't have a default value for the `UTF8*` and then it compiled. Sigh. Since SString does so many implicit conversions this was working but it needs to be `(SString::Utf8, "")` throughout.

Reply:
Ok. But what is this trying to achieve? Is this functional fix or a performance improvement?

Reply:
This ensures the `SString` is initialized to the proper encoding. That is why the `GetUTF8NoConvert()` is nice in this case. If we initialize to UTF8 and add anything other than UTF8, then an exception will be thrown.

Reply:
`SString` flips its internal encoding internally as needed. One does not initialize SString for a particular encoding.

I think these explicit Utf8 initializations can be deleted. They just make the code a bit bigger and slower, but otherwise does not achieve anything else.